rust/datafusion/src/optimizer/utils.rs

```diff
@@ -91,6 +91,8 @@ pub fn expr_to_column_names(expr: &Expr, accum: &mut HashSet<String>) -> Result<
         Expr::AggregateUDF { args, .. } => exprlist_to_column_names(args, accum),
         Expr::ScalarFunction { args, .. } => exprlist_to_column_names(args, accum),
         Expr::ScalarUDF { args, .. } => exprlist_to_column_names(args, accum),
+        // Between is rewritten in the physical plan
+        Expr::Between { .. } => Ok(()),
```

Comment:
I think column names should still be extracted here?

Reply:
Yes, you may be right. I am rewriting the Expr::Between into a set of Binary operations prior to that so I don't think this should ever be invoked.

Reply:
Maybe I can rewrite the expression in the Logical Plan (i.e. as early as possible) then the Optimizer and Physical plans are not required. I will refactor.

Reply:
@Dandandan added optimizer rules